Lib/email/message.py

```diff
@@ -286,8 +287,11 @@ def get_payload(self, i=None, decode=False):
         if i is not None and not isinstance(self._payload, list):
             raise TypeError('Expected list, got %s' % type(self._payload))
         payload = self._payload
+        cte = self.get('content-transfer-encoding', '')
+        if isinstance(cte, headerregistry.ContentTransferEncodingHeader):
+            cte = cte.cte
```

Comment:
Instead of an insinstance check, it is better to use a "quack like a duck" test here:
```
if hasattr(cte, 'cte'):
    cte = cte.cte
```

The reason for doing this is that we would like the Message and EmailMessage classes to remain agnostic about how the policy in effect does its work.  This means it should not know about the HeaderRegistry.  It is, however, reasonable to decide that the attributes that HeaderRegistry exposes (such as 'cte') should be the API that any policy should conform to if it provides header attributes at all.  Thus the duck-type test.  (And in the future, if we drop compat32, we might decide that all policies need to provide a minimum set of header attributes...but that's a future decision.)